### PR TITLE
static-code-analyzer use directory realtive to root pom

### PR DESCRIPTION
**Pull request by maggu2810**
```sh
pushd extensions/binding/org.eclipse.smarthome.binding.astro/; mvn clean install; popd
```

before:
```text
[INFO] BUILD FAILURE
...
[ERROR] Failed to execute goal org.openhab.tools:static-code-analysis:0.3.0:checkstyle (default) on project org.eclipse.smarthome.binding.astro: Unable to execute mojo: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: Unable to find suppressions file at location: /home/rathgeb/bin/pkgs/eclipse/esh/git/smarthome/extensions/binding/org.eclipse.smarthome.binding.astro/tools/static-code-analysis/checkstyle/suppressions.xml: Could not find resource '/home/rathgeb/bin/pkgs/eclipse/esh/git/smarthome/extensions/binding/org.eclipse.smarthome.binding.astro/tools/static-code-analysis/checkstyle/suppressions.xml'. -> [Help 1]
```

after:
```text
[INFO] BUILD SUCCESS
```

Fixes: https://github.com/eclipse/smarthome/issues/4230

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Interesting plugin. I guess we will need to request a build-time dependency CQ for it, before we can merge.

File: pom.xml [hunk 1]
```diff
@@ -71,6 +71,23 @@
   <build>
     <plugins>
       <plugin>
+        <groupId>org.commonjava.maven.plugins</groupId>
+        <artifactId>directory-maven-plugin</artifactId>
+        <version>0.1</version>
```
Comment (kaikreuzer) [inline pom.xml]: There is a version 0.2 available (see https://mvnrepository.com/artifact/org.commonjava.maven.plugins/directory-maven-plugin) - any special reason why you nonetheless chose 0.1?

**Comment by kaikreuzer** [comment]
I have created CQ 14162 for it.

File: pom.xml (hunk 1 above)
Comment (maggu2810) [reply pom.xml]: No special reason, I use the version 0.1 in several projects and it is doing its job.
It seems the version 0.2 has been released on July 31th, 2017 and I was not aware of.
If 0.2 is working as expected, let's use the newer one.

**Comment by maggu2810** [comment]
CQ has been approved but I don't see any version constrained in the CQ.
Doesn't the version matter?
Should I update this PR to use 0.2 and we are ready for merge?

**Comment by kaikreuzer** [comment]
It has only been PMC-approved, not yet closed by the IP team, so please wait for that.
Yes, for build-time dependencies, the version does not matter, so please update to 0.2 if that works as well.

**Comment by maggu2810** [comment]
> so please update to 0.2 if that works as well

Seems to do its job.

**Comment by kaikreuzer** [comment]
CQ has been approved!

**Pull request merged**